server/src/main/java/io/deephaven/server/console/python/DebuggingInitializer.java

```diff
@@ -0,0 +1,45 @@
+package io.deephaven.server.console.python;
+
+import io.deephaven.configuration.Configuration;
+import io.deephaven.util.thread.ThreadInitializationFactory;
+import org.jpy.PyLib;
+import org.jpy.PyModule;
+import org.jpy.PyObject;
+
+import java.io.Closeable;
+
+/**
+ * If python is configured as the script language for this server, ensures that threads which may invoke python code
+ * will be able to be debugged. If python is disabled, this does nothing.
+ */
+public class DebuggingInitializer implements ThreadInitializationFactory {
+    @Override
+    public Runnable createInitializer(Runnable runnable) {
+        if (!"python".equals(Configuration.getInstance().getStringWithDefault("deephaven.console.type", null))) {
+            // python not enabled, don't accidentally start it
+            return runnable;
+        }
+        DeephavenModule py_deephaven = (DeephavenModule) PyModule.importModule("deephaven")
```

Comment:
Note from @rcaudy : I haven't reviewed the bottom of this file past this point, or `py/server/deephaven/__init__.py` yet.